internal/controller/driver_controller.go

```diff
@@ -461,6 +462,7 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 								Name:            fmt.Sprintf("csi-%splugin", r.driverType),
 								Image:           r.images["plugin"],
 								ImagePullPolicy: imagePullPolicy,
+								SecurityContext: utils.If(r.logRotation(), utils.Privileged(), &corev1.SecurityContext{Privileged: ptr.To(false)}),
```

Comment:
You repeat the same `If` for SecurityContext on every/most container. 
In that case, please precalculate the value once above the deployment spec definition into a local variable and use that variable instead. You can take an example from the pre-calc for `imagePullPolicy`